corehq/ex-submodules/soil/progress.py

```diff
@@ -122,14 +122,14 @@ def get_task_status(task, is_multiple_download_task=False):
             pass
         else:
             result = task.result
-            if result and result.get('errors'):
-                failed = True
-                context_error = result.get('errors')
-            elif task.successful():
+            if task.successful():
                 is_ready = True
                 context_result = result and result.get('messages')
             elif result and isinstance(result, Exception):
                 context_error = six.text_type(result)
+            elif result and result.get('errors'):
+                failed = True
+                context_error = result.get('errors')
```

Comment:
Why are you reverting this?
Can just move this elif before the one above this to fix the issue caused?

Reply:
Because this is how it was working before I touched it. I think its better to keep it like this, document how to correctly mark errors and then apply that process to case importer and other places, instead of changing a core library without documentation

Reply:
Okay.